ts/src/gate.ts

```diff
@@ -4219,20 +4219,44 @@ export default class gate extends Exchange {
                 request['amount'] = this.amountToPrecision (symbol, amount);
             } else {
                 if (side === 'sell') {
-                    request['size'] = Precise.stringNeg (this.amountToPrecision (symbol, amount));
+                    request['size'] = this.parseNumber (Precise.stringNeg (this.amountToPrecision (symbol, amount)));
```

Comment:
we shouldn't use parseNumber inside the request because its return type is dynamic